classes/archiveduser.php

```diff
@@ -174,34 +170,29 @@ public function activate_me() {
     public function delete_me() {
         global $DB;
 
-        $user = new \core_user();
-        $user = $user->get_user($this->id);
+        $user = \core_user::get_user($this->id);
 
         if ($user != false and $user->deleted == 0 and !is_siteadmin($user)) {
 
             $transaction = $DB->start_delegated_transaction();
 
             // Deletes the records in both plugin tables.
-            if (!empty($DB->get_records('tool_cleanupusers', array('id' => $user->id)))) {
-                $DB->delete_records('tool_cleanupusers', array('id' => $user->id));
-            }
+            $DB->delete_records('tool_cleanupusers', array('id' => $user->id));
 
-            if (!empty($DB->get_records('tool_cleanupusers_archive', array('id' => $user->id)))) {
-                $DB->delete_records('tool_cleanupusers_archive', array('id' => $user->id));
-            }
+            $DB->delete_records('tool_cleanupusers_archive', array('id' => $user->id));
 
             // To secure that plugins that reference the user table do not fail create empty user with a hash as username.
             $newusername = hash('md5', $user->username);
 
             // Checks whether the username already exist (possible but unlikely).
-            if (empty($DB->get_record('user', array("username" => $newusername)))) {
+            if (!$DB->record_exists('user', array("username" => $newusername))) {
                 $user->username = $newusername;
                 user_update_user($user, false);
             } else {
                 // In the unlikely case that hash(username) exist in the table, while loop generates new username.
-                while (!empty($DB->get_record('user', array("username" => $newusername)))) {
+                while ($DB->record_exists('user', array("username" => $newusername))) {
                     $tempname = $newusername;
-                    $newusername = hash('sha256', $user->username . $tempname);
+                    $newusername = hash('md5', $user->username . $tempname);
```

Comment:
Is there a reason, why you have chosen a different hashing algorithm?

Reply:
In L. 185, the algorithm was already changed to md5 (maybe because it's faster?). It seems like it was forgotten here.

Reply:
This change was made due to the issue #5. sha256 generates to long user names.